src/transformers/pipelines/automatic_speech_recognition.py

```diff
@@ -136,6 +165,22 @@ def _sanitize_parameters(self, **kwargs):
         # No parameters on this pipeline right now
         return {}, {}, {}
 
+    def run_single(self, inputs, preprocess_params, forward_params, postprocess_params):
```

Comment:
Had to overload this method to allow for generator outputs from `self.preprocess()`. Maybe there's a better way.